src/lib/viewers/text/CSVViewer.js

```diff
@@ -77,6 +87,69 @@ class CSVViewer extends TextBaseViewer {
             .catch(this.handleAssetError);
     }
 
+    /**
+     * Checks for parse errors and if present triggers an error silently
+     * @param {Array} errors Array of errors from PapaParse parse results
+     * @return {void}
+     */
+    checkForParseErrors(errors = []) {
+        if (!errors.length) {
+            return;
+        }
+
+        const parseError = this.getWorstParseError(errors);
+
+        const error = new PreviewError(ERROR_CODE.PARSE_CSV, undefined, {
+            ...parseError,
+            silent: true,
+        });
+
+        this.triggerError(error);
+    }
+
+    /**
+     * Utility to sort the PapaParse errors by most significant and returning the first.
+     * The significance is defined as DELIMTER > QUOTES > FIELD_MISMATCH
+     * @param {Array} errors Array of errors from PapaParse parse results
+     * @return {Object} returns PapaParse error or undefined
+     */
+    getWorstParseError(errors = []) {
+        return errors.sort((a, b) => {
+            const { type: aType } = a;
+            const { type: bType } = b;
+
+            if (aType === PAPAPARSE_ERROR_TYPES.DELIMITER) {
```

Comment:
This could probably be simplified a bit with a priority map:

```
const ERROR_PRIORITY = {
    PAPAPARSE_ERROR_TYPES.DELIMITER: 0,
    PAPAPARSE_ERROR_TYPES.QUOTES: 1,
    PAPAPARSE_ERROR_TYPES.FIELD_MISMATCH: 2,
}

errors.sort((a, b) => {
    return ERROR_PRIORITY[b.type] - ERROR_PRIORITY[a.type];
});
```

Reply:
Good idea, i'll give it a try